src/service.cc

```diff
@@ -436,6 +470,10 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
             if (s.bad()) {
                 abort = true;
             }
+            if (residual_rec != nullptr) {
+                // We have a residuals record, write it to disk.
+                residualsFile->write_record(*residual_rec);
```

Comment:
We should not do this if `s.bad()` above, and we need to check whether it succeeds or fails.